components/application-gateway/internal/authorization/util/util.go

```diff
@@ -0,0 +1,21 @@
+package util
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/http"
+
+	"github.com/kyma-project/kyma/components/application-gateway/internal/httpconsts"
+)
+
+func AddBasicAuthHeader(request *http.Request, clientId, clientSecret string) {
+	basicAuthHeader := fmt.Sprintf("Basic %s", basicAuth(clientId, clientSecret))
+
+	request.Header.Set(httpconsts.HeaderAuthorization, basicAuthHeader)
+}
+
+func basicAuth(username, password string) string {
```

Comment:
Changing the name of this function to something more descriptive like `encodeBasicCredentials` would be a good move I bet